specviz/plugins/arithmetic/arithmetic_editor.py

```diff
@@ -0,0 +1,230 @@
+            
+            self.list_derived_components.takeTopLevelItem(index)
+
+    def showDialog(self):  
+        """Show arithmetic editor""" 
+        self.editor = Editor(self, parent=self)     
+
+    def find_matches(self, eq_name):
+        """Checks for matching names"""
+        matches = self.list_derived_components.findItems(eq_name, Qt.MatchExactly, 0)
+        matches = [item.text(0) for item in matches]
+        return matches
+    
+class Editor(QDialog):
+    tip_text = ("<b>Note:</b> Attribute names in the expression should be surrounded "
+                "by {{ }} brackets (e.g. {{{example}}}), and you can use "
+                "Numpy functions using np.&lt;function&gt;, as well as any "
+                "other function defined in your config.py file.<br><br>"
+                "<b>Example expressions:</b><br><br>"
+                "  - Subtract 10 from '{example}': {{{example}}} - 10<br>"
+                "  - Scale '{example}' to [0:1]: ({{{example}}} - np.min({{{example}}})) / np.ptp({{{example}}})<br>"
+                "  - Multiply '{example}' by pi: {{{example}}} * np.pi<br>"
+                "  - Use masking: {{{example}}} * ({{{example}}} &lt; 1)<br>")
+
+    placeholder_text = ("Type any mathematical expression here - "
+                        "you can include attribute names from the "
+                        "drop-down below by selecting them and "
+                        "clicking 'Insert'. Note: Arithmeric Editor "
+                        "must return a Spectrum1D object!")
+    
+    def __init__(self, equation_editor, label=None, equation=None, parent=None):
+        """Dialog where you specify equation names and expressions."""
+        super(Editor, self).__init__(parent)
+        loadUi(os.path.abspath(
+            os.path.join(os.path.dirname(__file__),
+                         ".", "equation_editor.ui")), self)
+    
+        self._equation_editor = equation_editor
+
+        if not self._equation_editor.model_items:
+            self.msgbox = QMessageBox.warning(self._equation_editor, "No Spectrum1D Objects", 
+                                "There is no data loaded into your SpecViz session!")
+        else:
+            self.is_addmode = label is None
+
+            # example=self._equation_editor.model_items
+            # self.label.setText(self.tip_text.format(example=example[0]))
+
+            if label is not None:
+                self.text_label.setText(label)
+                self.text_label.setDisabled(True)
+            else:
+                self.text_label.setPlaceholderText("New attribute name")
+            if equation is not None:
+                self.expression.insertPlainText(equation)
+            else:
+                self.expression.setPlaceholderText(self.placeholder_text)
+
+            # Set options for spec1d objects to select.        
+            self.combosel_data.addItems([item.name for item in self._equation_editor.model_items])
```

Comment:
This should be something like `self.combosel_data.setModel(self._equation_editor.hub.model)` ... That way the combo box is updated when the list of data items changes.

Reply:
Good point, I like this one way more.